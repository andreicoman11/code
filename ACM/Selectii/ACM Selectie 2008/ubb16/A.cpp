#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>
#include <map>
#include <list>
#include <cmath>
#include <algorithm>
#include <set>
#include <iomanip>
using namespace std;

ifstream fin ("A.in");
//ofstream fout ("A.out");

int N;

bool isproject(string);

int main()
{
	string project;
	getline(fin, project);
	while( project[0]!='0' )
	{
		vector<string> projects;
		map<string,int> ids;
		map<string,bool> students;
		projects.push_back(project);
		int cur = 0;
		string tmp;
		getline(fin, tmp);
		int v[100] = {0};
		while( tmp[0]!='1' )
		{
			//cout << tmp << " ";
			if( isproject(tmp) )
			{
				projects.push_back( tmp );
				cur++;
			}
			else
			{
				if( students[tmp] )
				{
					//cout << "student exists ";
					if( ids[tmp]!=cur && ids[tmp]!=-1 )
					{
						//cout << "will be removed ";
						v[ ids[tmp] ]--;
						ids[tmp]=-1;
					}
				}
				else
				{
					v[ cur ]++;
					ids[tmp] = cur;
					students[tmp] = true;
				}
			}
			//cout << cur << "\n";
			getline(fin, tmp);
		}

		/*cout << cur << "   ";
		for(int i=0; i<=cur; i++)
			cout << v[i] << " " << projects[i];*/

		bool ok = true;
		while(ok)
		{
			ok = false;
			for(int i=0; i<cur; i++)
				if( v[i]<v[i+1] || 
					(v[i]==v[i+1] && projects[i]>projects[i+1] ) )
				{
					ok = true;
					int aux = v[i];
					v[i] = v[i+1];
					v[i+1] = aux;
					tmp = projects[i];
					projects[i] = projects[i+1];
					projects[i+1] = tmp;
				}
		}

		for(int i=0; i<=cur; i++)
			cout << projects[i] << " " << v[i] << "\n";

		getline(fin, project);
	}
	return 0;
}

bool isproject(string t)
{
	for(int i=0; i<t.size(); i++)
		if( isalpha(t[i]) && islower(t[i]) ) return false;
	return true;
}