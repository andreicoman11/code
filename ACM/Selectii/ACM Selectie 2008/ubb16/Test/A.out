YOUBOOK 2
LIVESPACE BLOGJAM 1
UBQTS TXT 1
SKINUX 0
DESPERADOS 1
TRULY UGLY 0
TWIDDLERS 0
IEJHJBCBB 65
BHBGFIFIFD 61
BGBBEGDDHA 60
GEAJBFGJD 60
GBBFGBEEE 59
GJIGJDJGI 59
IIFEDAJGJ 59
CJBBGIACI 58
DGCCFJGHA 58
HDBFDJHJH 55
HGGCHAICC 55
BCJFIIHABF 53
JIACBFEFF 53
JIBACIFIG 53
BHHFGDIAJJ 50
BIDBJGCCFJ 50
JJEIICGGH 50
BIFEBCHIA 49
CAEHHAGIII 48
IEIBJCHDH 48
BHDFECECII 47
JCIDAHAFC 47
JEFCHJFDI 47
CACBDEHHHH 45
IDJJIJHJA 45
DDBGCCCIF 44
DIFAIABCH 44
BIJDBHFFI 42
DFJFDJGDA 42
HAGGIHAIH 42
BABAFGFCGE 39
BIAIBABAC 39
BJJFHDBCGI 39
BAAIIECEJ 38
BDBJBHHBFI 38
GICAJECGC 38
BBIJBJDIFA 37
BCHBHJAJJF 37
BDIIFDICHD 37
HHBFCCDGF 37
BEFFJBHIJG 36
FHIDBHAFH 36
GHHCACDHI 36
BHEIBFHIIB 34
BBCEBEGBIF 33
BDHIDFCDAB 33
BIBCAJEJCA 33
BJFAIGGCHD 33
DJADEIBGC 33
FAAIBAHHA 33
HADADACGB 33
BDBDBIBIAD 32
BGHBHHDFCH 32
BIBIDDCGB 32
DAIHICDA 32
CDDHACABJ 29
BABCBBIHIC 27
BAFFBDGGFF 27
JDAAEFAB 27
JDFIIEJDE 27
BIEGAGAAIF 25
CACIFJBFAI 25
JEIIABADJ 23
CBGHFEBCI 22
JEHHDDACD 22
BCBGJJJACH 20
EDHFJCJDE 20
CJIAIAHED 17
GJEFBIFIC 15
BFDHEHHCI 14
CBCEDFDEDB 14
BBBDICEIHF 13
BFDJAEIIAD 13
BCFCGFBC 11
BHCFCJCJBG 11
BJGEHEAFDJ 11
CFIDDECGI 11
BBGHJBIBAI 10
CAHIJBHBHG 10
DCGDEICFG 9
BBHABHEHGB 8
BDIAIAIBJI 8
GCICHJICD 8
BHDHBGGFFH 7
BHJFHFDDHD 7
BDCCFIHFCE 6
JCGADIGJ 6
BCIEAFEDIB 5
BIIJEJBCBF 5
BICDFIBEBB 4
FBIHBJEBI 4
CBEDDEGBJB 3
CGHBECGBB 3
BCDEDCJCCH 2
BDGIHJBBHH 2
IEIFGJBG 2
BFCBFACDHG 1
BFICFHCIIJ 1
BBGCGGAECF 0
BJDCHBGDCI 0
